crates/matrix-sdk/src/room/mod.rs

```diff
@@ -1704,7 +1706,30 @@ impl Room {
         self.send_state_event(RoomPowerLevelsEventContent::from(power_levels)).await
     }
 
-    async fn get_room_power_levels(&self) -> Result<RoomPowerLevels> {
+    /// Creates a set of power level changes based on the current power levels
+    /// of this room.
+    ///
+    /// This can be used to present existing settings to a user. To update the
+    /// power levels it is recommended to create a new `RoomPowerLevelChanges`,
+    /// set the values you would like to change and then call
+    /// `apply_power_level_changes` with the new settings.
+    pub async fn current_power_level_changes(&self) -> Result<RoomPowerLevelChanges> {
```

Comment:
I've a proposal for a new name to clarify things:

```suggestion
    pub async fn build_power_level_changes_from_current(&self) -> Result<RoomPowerLevelChanges> {
```

or another suggestion:

```suggestion
    pub async fn new_power_level_changes_from_current(&self) -> Result<RoomPowerLevelChanges> {
```

Thoughts?

Reply:
Ideally, we would want to build `RoomPowerLevelChanges` with its own constructor, like so:

```rust
RoomPowerLevelChanges::new()
```

And to build it from the current room:

```rust
RoomPowerLevelChanges::from(room)
```

It would be more Rust idiomatic and probably cleaner. What do you think?